src/plonk/mod.rs

```diff
@@ -317,10 +318,12 @@ impl<F: PrimeField> PlonkStructure<F> {
 
         let total_row = 1 << self.k;
 
+        let evaluator = GraphEvaluator::new(self.custom_gates_lookup_compressed.compressed());
         (0..total_row)
             .into_par_iter()
             .map(|row| {
-                data.eval(self.custom_gates_lookup_compressed.compressed(), row)
+                evaluator
+                    .evaluate(&data, row)
```

Comment:
This fn looks good. Have you tested this evaluation function e2e with this new code?

Reply:
This is literally the same code as it was, only without recreating the `GraphEvaluator` on each iteration. 

I'm running the e2e test right now, but it's unlikely that this version will create new problems